### PR TITLE
Mandatory and optional labels added to schema (fixes #520)

### LINKED ISSUE #520: Global ARIA labels for required and optional content
### Subject of the enhancement
Translatable labels for required or optional pages/articles/blocks/components.

### Expected behaviour
So we can better display this status. This was intended for menu items and page headers, but it should be usable in all content types.

### PR DESCRIPTION
**Pull request by zubairslamdien**
Fixes #520 

[//]: # (Link the PR to the original issue)

### New
* New "mandatory" and "optional" labels added to _ariaLabels in the course schema. Updates to the page template and Boxmenu will follow.

### CODE REVIEW

File: schema/course.model.schema [hunk 1]
```diff
@@ -338,6 +338,22 @@
                   "inputType": "Text",
                   "required": true,
                   "translatable": true
+                },
+                "mandatory": {
```
Comment (oliverfoster) [inline schema/course.model.schema:342]: mandatory vs required?

**Review by StuartNicholls — APPROVED** [review]

File: schema/course.model.schema (hunk 1 above)
Comment (StuartNicholls) [reply schema/course.model.schema:342]: @oliverfoster I think mandatory is the most commonly used. I guess it's a stronger word? I'll go with consensus though

Comment (oliverfoster) [reply schema/course.model.schema:342]: Stronger in what sense? More characters and syllables? `required` has a higher scrabble score at 18 vs `mandatory`'s 15.

Comment (StuartNicholls) [reply schema/course.model.schema:342]: If words had muscles, Mandatory would have bigger ones

Comment (oliverfoster) [reply schema/course.model.schema:342]: If we go for `"compulsory": "Compulsory"` we could compel people to do the learning rather than just mandate it?

Comment (kirsty-hames) [reply schema/course.model.schema:342]: I would have thought "required" to be the common use case but I have seen both use cases. The `default` can easily be amended per course when needed.

**Review by kirsty-hames — APPROVED** [review]

**Review by oliverfoster — APPROVED** [review]

**Review by kirsty-hames — APPROVED** [review]

**Review by cahirodoherty-learningpool — APPROVED** [review]
👀

**Pull request merged**